xmind/tests/test_position.py

```diff
@@ -0,0 +1,111 @@
+
+    def test_init(self):
+        _workbook_mixin_element_init = self._init_patch_with_name(
+            '_mixin_init', 'xmind.core.position.WorkbookMixinElement.__init__', autospec=True)
+
+        _obj1 = PositionElement()
+        _workbook_mixin_element_init.assert_called_with(_obj1, None, None)
+
+        _obj2 = PositionElement(1)
+        _workbook_mixin_element_init.assert_called_with(_obj2, 1, None)
+
+        _obj3 = PositionElement(None, 1)
+        _workbook_mixin_element_init.assert_called_with(_obj3, None, 1)
+
+        _obj4 = PositionElement(1, 2)
+        _workbook_mixin_element_init.assert_called_with(_obj4, 1, 2)
+
+        _obj5 = None
+        with self.assertRaises(TypeError) as _ex:
+            _obj5 = PositionElement(1, 2, 3)
+
+        self.assertEqual(None, _obj5)
+        self.assertTrue(_ex.exception.args[0].find(
+            "__init__() takes") != -1)
+        self.assertEqual(_workbook_mixin_element_init.call_count, 4)
+
+    def test_excessive_parameters(self):
+        _workbook_mixin_element_init = self._init_patch_with_name(
+            '_mixin_init', 'xmind.core.position.WorkbookMixinElement.__init__')
+        _get_attribute = self._init_patch_with_name(
+            '_get_attribute', 'xmind.core.position.PositionElement.getAttribute')
+        _set_attribute = self._init_patch_with_name(
+            '_set_attribute', 'xmind.core.position.PositionElement.setAttribute')
+
+        _element = PositionElement()
+        _workbook_mixin_element_init.assert_called_once_with(None, None)
+
+        _parameters = [
+            ('getX', 0),
+            ('getY', 0),
+            ('setX', 1),
+            ('setY', 1)
+        ]
+
+        for pair in _parameters:
+            with self.subTest(pair=pair):
+                self._test_method_by_excessive_parameters(pair, _element)
+
+        _get_attribute.assert_not_called()
+        _set_attribute.assert_not_called()
+
+    def test_getX(self):
+        _workbook_mixin_element_init = self._init_patch_with_name(
+            '_mixin_init', 'xmind.core.position.WorkbookMixinElement.__init__')
+
+        _element = PositionElement()
+        with patch.object(_element, 'getAttribute') as _get_attribute:
+            _get_attribute.return_value = 10
+            self.assertEqual(10, _element.getX())
+
+        _workbook_mixin_element_init.assert_called_once_with(None, None)
+        _get_attribute.assert_called_once_with("svg:x")
+
+    def test_getY(self):
+        _workbook_mixin_element_init = self._init_patch_with_name(
+            '_mixin_init', 'xmind.core.position.WorkbookMixinElement.__init__')
+
+        _element = PositionElement()
+        with patch.object(_element, 'getAttribute') as _get_attribute:
+            _get_attribute.return_value = 11
+            self.assertEqual(11, _element.getY())
+
+        _workbook_mixin_element_init.assert_called_once_with(None, None)
+        _get_attribute.assert_called_once_with("svg:y")
```

Comment:
This is questionable.

Why do I use just words instead of constants?
===

We can have 2 approaches:
----
1) we pass const.WHATEVER in tests and whenever someone modifes this value in const file we will not break our tests, but if it was done unitentionally then our tests become false positive which is not so good.
2) we pass raw value and whenver someone modifes unitentionally consts we imediately catch it. Drawback here is that we have to go and modify all tests where we used raw value before. And this approach as for me is more controllable although it requires more work later. But const means no changes!

Reply:
as for me it is better to have test, that checks all constant values in const.py and use const.WHATEVER in all other tests. in such case we will catch unintentional modification    of the const.py but will not have to update all other tests in case we need to change something in const.py.

Reply:
ok, agree! will convert to const.WHATEVER in my tests